File: API_R_01Agosto2024/XamaCardoso/q2_hotel.py
def get_number_in_range(minimo, maximo, text):
    n = int(input(text));
    if n == 0:
        return n;
    
    if n < minimo or n > maximo:
        print(f"Número inválido!! Digite um valor entre {minimo} e {maximo}!!");
        get_number_in_range(minimo,maximo,text);
    
    return n;

def formar_quarto(num):
    dias = int(input("Quantas noites é a reserva? "));
    preco = 200 + (80*num) * dias;
    taxajuros = 0.105;
    precojuros = preco + preco*taxajuros;
    precopix = preco - preco*0.05;
    print(f"""
============= NOVO QUARTO =====================
Pessoas                         : {num} pessoas
Preço via PIX (5% de desconto)  : R$ {precopix:.2f}
Preço em até 3x (sem Juros)     : R$ {preco:.2f}
Preço de 4x a 12x (com Juros)   : R$ {precojuros:.2f}
===============================================
""");


def main():
    qtd_pessoas = 0
    idade = 1
    while idade != 0:
        idade = get_number_in_range(18, 100, "Digite a idade do cliente: ")
        if idade != 0:
            qtd_pessoas += 1
            if qtd_pessoas == 4:
                formar_quarto(qtd_pessoas);
                qtd_pessoas = 0;

    if qtd_pessoas > 0:
        formar_quarto(qtd_pessoas);



main()

File: API_R_01Agosto2024/FelipeMadson/q2_hotel.js
import { question } from 'readline-sync';

function calculateRoomPrice(guests) {
  const basePrice = 200;
  const priceIncrement = 80;
  return Math.min(basePrice + (guests - 1) * priceIncrement, 440);
}

function main() {
  let totalGuests = 0;
  let rooms = [];
  let currentRoom = [];

  while (true) {
    const input = question("Digite a idade do hóspede (0 para novo quarto, 00 para encerrar): ");

    if (input === '00') {
      // Dois zeros consecutivos, encerramos o programa
      break;
    }

    if (input === '0') {
      // Um zero, finalizamos o quarto atual e começamos um novo
      if (currentRoom.length > 0) {
        rooms.push(currentRoom);
        currentRoom = [];
        console.log("Novo quarto iniciado.");
      } else {
        console.log("Quarto atual está vazio. Digite uma idade válida.");
      }
      continue;
    }

    const age = parseInt(input);

    if (age < 18 || age > 100 || isNaN(age)) {
      console.log("Idade inválida. Por favor, digite uma idade entre 18 e 100 anos.");
      continue;
    }

    currentRoom.push(age);
    totalGuests++;

    if (currentRoom.length === 4) {
      rooms.push(currentRoom);
      currentRoom = [];
      console.log("Quarto lotado. Novo quarto iniciado automaticamente.");
    }
  }

  if (currentRoom.length > 0) {
    rooms.push(currentRoom);
  }

  if (rooms.length === 0) {
    console.log("Nenhuma reserva foi feita.");
    return;
  }

  console.log(`\nTotal de quartos reservados: ${rooms.length}`);
  console.log(`Total de hóspedes: ${totalGuests}`);

  let totalDailyPrice = 0;
  rooms.forEach((room, index) => {
    const roomPrice = calculateRoomPrice(room.length);
    totalDailyPrice += roomPrice;
    console.log(`Quarto ${index + 1}: ${room.length} hóspedes, R$ ${roomPrice.toFixed(2)}/dia`);
  });

  const nights = parseInt(question("\nPara quantas noites será a reserva? "));

  const totalPrice = totalDailyPrice * nights;
  console.log(`\nValor total por dia: R$ ${totalDailyPrice.toFixed(2)}`);
  console.log(`Valor total da reserva para ${nights} dias: R$ ${totalPrice.toFixed(2)}`);

  const pixDiscount = totalPrice * 0.05;
  console.log(`\nDesconto de 5% para pagamento via PIX: R$ ${pixDiscount.toFixed(2)}`);
  console.log(`Valor total com desconto PIX: R$ ${(totalPrice - pixDiscount).toFixed(2)}`);

  console.log("\nOpções de pagamento no cartão de crédito:");
  console.log("- Em até 3x sem juros");
  const creditInterest = totalPrice * 0.105;
  console.log(`- De 4x a 12x com juros fixos simples de 10,5%: R$ ${(totalPrice + creditInterest).toFixed(2)}`);
}

main();